pkg/cluster/keyvaults.go

```diff
@@ -0,0 +1,77 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/ARO-RP/pkg/util/stringutils"
+	"github.com/Azure/ARO-RP/pkg/util/uuid"
+	"github.com/Azure/azure-sdk-for-go/services/keyvault/mgmt/2019-09-01/keyvault"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+func (m *manager) createKeyvault(ctx context.Context) error {
+	// We want random characters from end of infraID to help ensure KV name uniqueness
+	infraID := m.doc.OpenShiftCluster.Properties.InfraID
+	suffixMaxLength := 20
+	var suffix string
+
+	// This runs after ensureInfraID bootstrap step, but just in case
+	if infraID == "" {
+		suffix = m.doc.OpenShiftCluster.Name
+	} else {
+		suffix = infraID
+	}
```

Comment:
```suggestion
        suffix := infraID
        
	// This runs after ensureInfraID bootstrap step, but just in case
	if infraID == "" {
		suffix = m.doc.OpenShiftCluster.Name
	} 
```